streams/client/client_test.go

```diff
@@ -164,3 +164,133 @@ func TestClient(t *testing.T) {
 		cd.Close()
 	}
 }
+
+func TestClientMultiSend(t *testing.T) {
+	logging.SetLogLevel("*", "error")
+
+	ctx, ctxC := context.WithCancel(context.Background())
+	defer ctxC()
+
+	rand.Seed(time.Now().UnixNano())
+
+	var n int
+	for n < 25565 || n > 40000 {
+		n = rand.Intn(100000)
+	}
+
+	p1, err := peer.New( // provider
+		ctx,
+		nil,
+		keypair.NewRaw(),
+		nil,
+		[]string{fmt.Sprintf("/ip4/127.0.0.1/tcp/%d", n)},
+		nil,
+		true,
+		false,
+	)
+	if err != nil {
+		t.Errorf("Peer creation returned error `%s`", err.Error())
+		return
+	}
+	defer p1.Close()
+
+	svr1, err := peerService.New(p1, "hello", "/hello/1.0")
+	if err != nil {
+		t.Errorf("Service creation returned error `%s`", err.Error())
+		return
+	}
+	defer svr1.Stop()
+	err = svr1.Define("hi", func(context.Context, streams.Connection, command.Body) (cr.Response, error) {
+		return cr.Response{"message": "HI"}, nil
+	})
+	if err != nil {
+		t.Error(err)
+		return
+	}
+
+	p2, err := peer.New( // provider
+		ctx,
+		nil,
+		keypair.NewRaw(),
+		nil,
+		[]string{fmt.Sprintf("/ip4/127.0.0.1/tcp/%d", n)},
+		nil,
+		true,
+		false,
+	)
+	if err != nil {
+		t.Errorf("Peer creation returned error `%s`", err.Error())
+		return
+	}
+	defer p2.Close()
+
+	svr2, err := peerService.New(p2, "hello", "/hello/1.0")
+	if err != nil {
+		t.Errorf("Service creation returned error `%s`", err.Error())
+		return
+	}
+	defer svr2.Stop()
+	err = svr2.Define("hi", func(context.Context, streams.Connection, command.Body) (cr.Response, error) {
+		return cr.Response{"message": "HI"}, nil
+	})
+	if err != nil {
+		t.Error(err)
+		return
+	}
+
+	p3, err := peer.New( // consumer
+		ctx,
+		nil,
+		keypair.NewRaw(),
+		nil,
+		[]string{fmt.Sprintf("/ip4/127.0.0.1/tcp/%d", n+1)},
+		nil,
+		true,
+		false,
+	)
+	if err != nil {
+		t.Errorf("Ping test returned error `%s`", err.Error())
+		return
+	}
+	defer p3.Close()
+
+	err = p3.Peer().Connect(ctx, peercore.AddrInfo{ID: p1.ID(), Addrs: p1.Peer().Addrs()})
+	if err != nil {
+		t.Errorf("Connect to peer %v returned `%s`", p1.Peer().Addrs(), err.Error())
+		return
+	}
+
+	err = p3.Peer().Connect(ctx, peercore.AddrInfo{ID: p2.ID(), Addrs: p2.Peer().Addrs()})
+	if err != nil {
+		t.Errorf("Connect to peer %v returned `%s`", p2.Peer().Addrs(), err.Error())
+		return
+	}
+
+	// discover
+	cd, err := New(ctx, p3, nil, "/hello/1.0", 2, 2)
+	if err != nil {
+		t.Errorf("Client creation returned error `%s`", err.Error())
+		return
+	} else {
+		res, errs, err := cd.MultiSend("hi", command.Body{}, 2)
+		if err != nil {
+			t.Errorf("Sending command returned error `%s`", err.Error())
+			return
+		}
+
+		if len(res) != 2 && len(errs) == 0 {
+			t.Errorf("MultiSending command failed R=%d, E=%d", len(res), len(errs))
+			return
+		}
+
+		for p, r := range res {
+			if r["message"] != "HI" {
```

Comment:
if val, ok := 
its just a test, but still should try to avoid panics if something goes wrong

Reply:
if val does not exist in a map the nil value of the val type is returned. here it's ""